azure/application-gateway/variables.tf

```
@@ -0,0 +1,243 @@
+    frontend_ip_configuration_name = optional(string, "PublicFrontend")
+    frontend_port_name             = optional(string, "Http")
+    protocol                       = optional(string, "Http")
+    host_name                      = optional(string)
+    host_names                     = optional(list(string))
+    ssl_certificate_name           = optional(string)
+  }))
+  default = [{
+    name = "Default"
+  }]
+}
+variable "ssl_certificates" {
+  description = "List of SSL Certs"
+  type = list(object({
+    name                = string
+    data                = optional(string)
+    password            = optional(string)
+    key_vault_secret_id = optional(string)
+  }))
+  default = []
+  validation {
+    condition     = try(alltrue([for cert in var.ssl_certificates : (cert.key_vault_secret_id != null || (cert.data != null && cert.password != null))]), true)
+    error_message = "Each certificate must specify either data and password or key_vault_secret_id."
+  }
+}
+
+variable "trusted_root_certificate" {
+  description = "List of SSL Certs"
+  type = list(object({
+    name = string
+    data = string
+  }))
+  default = []
+}
+
+variable "url_path_maps" {
+  description = "List of Path Maps"
+  type = list(object({
+    name                               = string
+    default_backend_address_pool_name  = string
+    default_backend_http_settings_name = optional(string, "Default")
+    path_rule = list(object({
+      name                       = string
+      paths                      = list(string)
+      backend_address_pool_name  = string
+      backend_http_settings_name = optional(string, "Default")
+    }))
+  }))
+  default = []
+}
+variable "request_routing_rules" {
+  description = "List of Routing Rules"
+  type = list(object({
+    name                        = string
+    rule_type                   = optional(string, "PathBasedRouting")
+    http_listener_name          = string
+    backend_address_pool_name   = optional(string, "Default")
+    backend_http_settings_name  = optional(string)
+    url_path_map_name           = optional(string)
+    priority                    = optional(number, 100)
+    redirect_configuration_name = optional(string)
+  }))
+}
+
+variable "redirect_configurations" {
+  description = "List of Redirection Configurations"
+  type = list(object({
+    name                 = string
+    redirect_type        = optional(string, "Permanent")
+    target_listener_name = optional(string)
+    target_url           = optional(string)
+    include_path         = optional(bool, true)
+    include_query_string = optional(bool, true)
+  }))
+  default = []
+}
+
+variable "identity_ids" {
+  description = "List of potential UserAssigned identities"
+  type        = list(string)
+  default     = null
+}
+
+variable "tags" {
+  description = "Resource Tags"
+  type        = map(string)
+  default     = null
+}
+
+variable "probe" {
+  description = "List of Routing Rules"
```

Correction to resource name

```suggestion
  description = "List of Probes"
```